### DISCUSSION TITLE
dial tcp: lookup proxy.golang.org i/o timeout

### DISCUSSION THREAD

**Discussion by sondip121** [projectdiscovery/httpx#658] [General] [answered]
![Screenshot from 2022-06-05 21-39-24](https://user-images.githubusercontent.com/73928464/172063302-8e969c2f-84ea-4bec-ba83-2d46c4ddb19f.png)

**Comment by Mzack9999** [comment]
It seems like your box has network connectivity issues. Please ensure that your box network is configured properly. A correct configuration should allow you to connect directly to `proxy.golang.org`:
```console
curl https://proxy.golang.org`
```

**Reply by nnmin-aws** [reply]
curl https://proxy.golang.org
curl: (6) Could not resolve host: proxy.golang.org

**Comment by psks1994** [comment, marked as answer]
Faced same issue, resolved using below
export GOPROXY=direct

**Reply by merchantli** [reply]
Where to add？

**Reply by duorme** [reply]
in the terminal, just run this cmd

**Comment by JingChen23** [comment]
I solved the issue by `systemctl restart docker` on CentOS 7.

Adding `export GOPROXY=direct` couldn't solve my problem.

**Reply by rahmaneffendi** [reply]
Thanks, it help me a lot

**Reply by pablodz** [reply]
restart docker worked, thanks

**Comment by meispi** [comment]
If someone is facing this issue on Windows try adding GOPROXY=direct in your environment variables as mentioned by @psks1994

**Comment by nccapo** [comment]
In my case `GOPROXY=direct` doesn't work. so i find solution, just export `export GO111MODULE=on` and `export GOPROXY=https://goproxy.cn`.
source: [https://goproxy.cn/](url)
Or you just need to clear Go cache, use command: `go clean -cache -modcache -i -r` after that, try to run your command again

**Reply by niharikaajain30** [reply]
export GO111MODULE=on and export GOPROXY=https://goproxy.cn worked for me on MAC.
